packages/bpk-component-autosuggest/src/BpkAutosuggestV2/BpkAutosuggest.tsx

```diff
@@ -249,20 +308,25 @@ const BpkAutosuggest = forwardRef<HTMLInputElement, BpkAutoSuggestProps<any>>(
         });
 
         if (type === useCombobox.stateChangeTypes.InputChange) {
-          if (newInputValue?.length > 0) {
+          if (newInputValue && newInputValue.length > 0) {
             if (newIsOpen) {
               onSuggestionsFetchRequested(newInputValue);
             }
           } else {
+            // Clear old suggestions before requesting defaults
+            onSuggestionsClearRequested?.();
             onSuggestionsFetchRequested('');
           }
         }
       },
       onSelectedItemChange(changes) {
         const { selectedItem } = changes;
         if (selectedItem) {
-          setInputValue(getSuggestionValue(selectedItem));
+          const newValue = getSuggestionValue(selectedItem);
+          setInputValue(newValue);
```

Comment:
the `newValue` updated, should `inputValue` in [L330](https://github.com/Skyscanner/backpack/pull/4043/files#diff-6e2a8f4a69f55e0360c7826245e732644a4d5677bdec98213315f664172a5651R330)  `onSuggestionSelected` also updated?